packages/cms/src/components/editors/SimpleVisualizationEditor.jsx

```diff
@@ -136,12 +174,52 @@ class SimpleVisualizationEditor extends Component {
     if (field === "type") {
       this.setState({object}, this.rebuild.bind(this));
     }
-    // Otherwise they are just changing a drop-down field and we need only recompile the code above.
+    // If the user is changing an x or y field, implicitly reset the label for the axis.
+    else if (field === "x" || field === "y") {
+      object[`${field}Config.title`] = e.target.value;
+      this.setState({object}, this.compileCode.bind(this));
+    }
+    // Otherwise they are just changing a drop-down field
     else {
       this.setState({object}, this.compileCode.bind(this));
     }
   }
 
+  onChangeFormatter(field, e) {
+    const {object} = this.state;
+    if (!object.formatters) object.formatters = {};
+    if (e.target.value === "manual-none") {
+      delete object.formatters[field];
+    }
+    else {
+      object.formatters[field] = e.target.value;
+    }
+    this.setState({object}, this.compileCode.bind(this));
+  }
+
+  getOptionList(method) {
+    const {payload} = this.state;
+    const firstObj = payload && payload.data && payload.data[0] ? payload.data[0] : {};
+    const isID = d => d.includes("ID ") || d.includes(" ID");
+    const allFields = Object.keys(firstObj);
+    const plainFields = Object.keys(firstObj).filter(d => !isID(d));
+    const idFields = Object.keys(firstObj).filter(d => isID(d));
```

Comment:
Instead of calculating `Object.keys(firstObj)` three times, just reference `allFields`.